src/app/modules/webhook/webhook.consumer.ts

```diff
@@ -0,0 +1,247 @@
+import aws from 'aws-sdk'
+import https from 'https'
+import mongoose from 'mongoose'
+import { errAsync, okAsync, ResultAsync } from 'neverthrow'
+import { Consumer } from 'sqs-consumer'
+
+import { SubmissionWebhookInfo } from '../../../types'
+import config from '../../config/config'
+import { createLoggerWithLabel } from '../../config/logger'
+import { getEncryptSubmissionModel } from '../../models/submission.server.model'
+import { transformMongoError } from '../../utils/handle-mongo-error'
+import { PossibleDatabaseError } from '../core/core.errors'
+import { SubmissionNotFoundError } from '../submission/submission.errors'
+
+import {
+  WebhookNoMoreRetriesError,
+  WebhookPushToQueueError,
+  WebhookRetriesNotEnabledError,
+  WebhookValidationError,
+} from './webhook.errors'
+import { WebhookQueueMessage } from './webhook.message'
+import { WebhookProducer } from './webhook.producer'
+import * as WebhookService from './webhook.service'
+import { isSuccessfulResponse } from './webhook.utils'
+
+const logger = createLoggerWithLabel(module)
+const EncryptSubmission = getEncryptSubmissionModel(mongoose)
+
+/**
+ * Starts polling a queue for webhook messages.
+ * @param queueUrl URL of queue from which to consume messages
+ * @param producer Producer which can be used to enqueue messages
+ */
+export const startWebhookConsumer = (
+  queueUrl: string,
+  producer: WebhookProducer,
+): void => {
+  const app = Consumer.create({
+    queueUrl,
+    region: config.aws.region,
+    handleMessage: createWebhookQueueHandler(producer),
+    // By default, the default Node.js HTTP/HTTPS SQS agent
+    // creates a new TCP connection for every new request.
+    // In production, pass an SQS instance to avoid the cost
+    // of establishing new connections.
+    sqs: config.isDev
+      ? undefined
+      : new aws.SQS({
+          region: config.aws.region,
+          httpOptions: {
+            agent: new https.Agent({
+              keepAlive: true,
+            }),
+          },
+        }),
+  })
+
+  app.on('error', (error, message) => {
+    logger.error({
+      message:
+        'Webhook consumer encountered error while interacting with queue',
+      meta: {
+        action: 'startWebhookConsumer',
+        message,
+      },
+      error,
+    })
+  })
+
+  app.start()
```

Comment:
if you check out the [implementation](https://github.com/bbc/sqs-consumer/blob/master/src/consumer.ts), this works because the consumer calls a `setTimeout` when we do `app.start()`, and again whenever it polls the queue. so this kind of works like a promise in that the next action has already been scheduled in the event loop when the function returns.